uaclient/data_types.py

```diff
@@ -36,6 +42,13 @@ def __init__(self, err: IncorrectTypeError, key: str):
         )
 
 
+class IncorrectEnumValueError(exceptions.UserFacingError):
```

Comment:
I think this needs to extend IncorrectTypeError to be caught properly by the recursive parsing functions